sirepo/pkcli/service.py

```diff
@@ -67,41 +70,68 @@ def _handle_signals(signums):
         finally:
             [signal.signal(x[0], x[1]) for x in o]
 
+    def _safe_kill_process(proc):
+        try:
+            proc.terminate()
+            proc.wait(1)
+        except (
+            ProcessLookupError,
+            ChildProcessError,
+            psutil.NoSuchProcess,
+        ):
+            pass
+        except (psutil.TimeoutExpired, subprocess.TimeoutExpired):
+            proc.kill()
+
     def _kill(*args):
         for p in processes:
             try:
-                p.terminate()
-                p.wait(1)
-            except (ProcessLookupError, ChildProcessError):
-                continue
-            except subprocess.TimeoutExpired:
-                p.kill()
-
-    def _start(service):
-        c = ["pyenv", "exec", "sirepo"]
-        c.extend(service)
+                for c in list(psutil.Process(p.pid).children(recursive=True)):
+                    _safe_kill_process(c)
+            except Exception:
+                # need to ignore exceptions while converting process children
+                # to a list so that the parent is still terminated
+                pass
+            _safe_kill_process(p)
+
+    def _start(service, extra_environ, cwd=".", prefix=("pyenv", "exec", "sirepo")):
+        env = PKDict(os.environ).pkmerge(extra_environ)
```

Comment:
Yes, this is what I was thinking about. pkmerge is not what we typically do for this. pkupdate is what we would want here, since these are flat values. The temporary is unnecessary, however.

Reply:
How would this work since extra_environ is also a PKDict

Reply:
Or should it be `extra_environ.pkupdate(os.environ)`?

Reply:
No, this would do the wrong thing. extra_environ overrides os.environ, always.

Reply:
Ok, I think the current version should resolve this